GLTFSDK/Source/GLTFResourceReader.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <GLTFSDK/GLTFResourceReader.h>
+
+using namespace Microsoft::glTF;
+
+namespace
+{
+	// Conversions of normalized component types to/from floats are explicitly defined in the 2.0 spec
+	inline float ComponentToFloat(const int8_t w)  { return std::max(static_cast<float>(w) / 127.0f, -1.0f); }
+	inline float ComponentToFloat(const uint8_t w) { return static_cast<float>(w) / 255.0f; }
+	inline float ComponentToFloat(const int16_t w) { return std::max(static_cast<float>(w) / 32767.0f, -1.0f); }
+	inline float ComponentToFloat(const uint16_t w){ return static_cast<float>(w) / 65535.0f; }
+
+	template<typename T>
+	std::vector<float> DecodeFloats(const Document& doc, const GLTFResourceReader& reader, const Accessor& accessor)

```

Comment:
Two comments of DecodeFloats:

1. Maybe rename this to DecodeToFloat so it's clearer that it converts to floats, not from them?
2. The use of reserve() and back_inserter is unnecessary, the code could be:

std::vector<T> rawData = reader.ReadBinaryData<T>(doc, accessor);

if (accessor.normalized)
{
	std::vector<float> decodedData(rawData.size());

	std::transform(
		rawData.begin(),
		rawData.end(),
		decodedData.begin(),
		static_cast<float(*)(const T)>(&ComponentToFloat));

	return decodedData;
}
else
{
	return std::vector<float>(rawData.begin(), rawData.end());
}

Reply:
Is there a specific reason why you recommend avoiding back_inserter here? Ideally I would use a raw for loop here, but I decided against rewriting this code at the time. For back_inserter specifically,

1. This might be faster than your version because it doesn't require memsetting the entire vector to zero (will be important for large vectors, probably unimportant for small)
3. This might be substantially faster than your version because it allows to inline the lambda whereas your version doesn't, so each call to ComponentToFloat requires a `call`.

I haven't compared the performance here, but I don't want to blindly change this code to the version you proposed because of the concerns I listed.

Reply:
1. I had assumed that repeated calls to push_back would end up having a greater overhead than the zero-initialization but I'm not so sure now. Without any benchmarking it's hard to say which approach is actually quicker! Feel free to leave the back_inserter here.

2. yep, best to revert to the lambda (although the explicit return type can be removed).

Reply:
For now I changed this to use a raw for loop since back_inserter doesn't seem to be used elsewhere in the code and reserve+push+back matches the pattern seen in various implementations in MeshPrimitiveUtils